logger/wandb_logger.py

```diff
@@ -0,0 +1,180 @@
+
+        if predictions.has("pred_masks"):
+            masks = predictions.pred_masks.cpu().numpy()
+            masks = [
+                GenericMask(x, self.image.shape[0], self.image.shape[1]) for x in masks
+            ]
+        else:
+            masks = None
+
+        if boxes is not None:
+            boxes_data = []
+            for i, box in enumerate(boxes):
+                if scores[i] > self.conf_threshold:
+                    pred_class = int(classes[i]) + 1
+                    caption = (
+                        f"{pred_class}"
+                        if not self.class_names
+                        else self.class_names[pred_class]
+                    )
+                    boxes_data.append(
+                        {
+                            "position": {
+                                "minX": box[0],
+                                "minY": box[1],
+                                "maxX": box[2],
+                                "maxY": box[3],
+                            },
+                            "class_id": pred_class,
+                            "box_caption": "%s %.3f" % (caption, scores[i]),
+                            "scores": {"class_score": scores[i]},
+                            "domain": "pixel",
+                        }
+                    )
+            if boxes_data:
+                boxes = {
+                    "prediction": {
+                        "box_data": boxes_data,
+                        "class_labels": self.class_names,
+                    }
+                }
+            else:
+                boxes = None
+        if masks is not None:
+            final_mask = np.zeros(
+                (self.image.shape[0], self.image.shape[1]), dtype=np.uint8
+            )
+            for i, mask in enumerate(masks):
+                pred_mask = mask.mask
+                pred_class = int(classes[i]) + 1
+                final_mask = np.ma.array(final_mask, mask=pred_mask)
+                final_mask = final_mask.filled(pred_class)
+            final_mask = final_mask.astype(np.uint8)
+            masks = {
+                "prediction": {
+                    "mask_data": final_mask,
+                    "class_labels": self.class_names,
+                }
+            }
+        return {
+            "data_or_path": self.image_path,
+            "boxes": boxes,
+            "masks": masks,
+            "classes": self.class_set,
+        }
+
+
+class WandbInferenceLogger:
+    """
+    Logs inference images and predictions to wandb.
+    Currently, supports bounding boxes and instance segmentation.
+    """
+
+    def __init__(
+        self,
+        wandb_entity: str = None,
+        wandb_project: str = None,
+        run_name: str = None,
+        class_names: Dict[int, str] = None,
+        conf_threshold: float = 0.7,
+        config=None,
+    ):
+        if not is_wandb_available():
+            raise ImportError("Please install it using 'pip install wandb'.")
```

Comment:
Can you replace the message with the below:

"Please install wandb using 'pip install wandb --upgrade'"

Just to make it explicit that wandb is the import error, and also to try build the habit for people to use --upgrade